### PR TITLE
test: add unit test about jobset controller

### PR DESCRIPTION
**Pull request by googs1025**
Hello everyone, when I was studying the project, I found that the degree of unit testing was a bit low. I tried to add some unit testing code, hoping to help the project.
Add some unit tests about jobset controller. Improve overall unit test coverage

### CODE REVIEW

**Review by danielvegamyhre — CHANGES_REQUESTED** [review]

File: pkg/controllers/jobset_controller_test.go [hunk 1]
```diff
@@ -1150,3 +1154,733 @@ func makeJob(args *makeJobArgs) *testutils.JobWrapper {
...
+					Succeeded: 0,
+				},
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			r := makeJobSetReconciler()
+			actual := r.createJobs(context.TODO(), tc.js, &tc.jobs, tc.replicatedJobStatus)
+			if diff := cmp.Diff(tc.expected, actual); diff != "" {
```
[the remark this answers is not in the document]
Comment (googs1025) [reply pkg/controllers/jobset_controller_test.go:1269]: Thank your review. Indeed, in the test cases, the robustness of the code was not taken into consideration. At that time, I only focused on whether the methods had exceptions.

[the remark this answers is not in the document]
Comment (googs1025) [reply pkg/controllers/jobset_controller_test.go:1269]: I will take your suggestions into account and make code modifications. Thank you for your advice.

**Comment by googs1025** [comment]
@danielvegamyhre Hello, thank you for your suggestion. What I'm thinking is that currently, there are quite a few methods that need unit testing. Can I raise an issue to track them and submit multiple pull requests? This way, it will be easier for the reviewers as well. I will close this pull request and submit a new issue to track the progress.

**Comment by danielvegamyhre** [comment]
> @danielvegamyhre Hello, thank you for your suggestion. What I'm thinking is that currently, there are quite a few methods that need unit testing. Can I raise an issue to track them and submit multiple pull requests? This way, it will be easier for the reviewers as well. I will close this pull request and submit a new issue to track the progress.

Yes that would be great

**Pull request closed without merging**